pallets/msa/src/lib.rs

```diff
@@ -533,23 +538,31 @@ pub mod pallet {
 		#[pallet::weight(T::WeightInfo::add_key_to_msa())]
 		pub fn add_key_to_msa(
 			origin: OriginFor<T>,
-			key: T::AccountId,
-			proof: MultiSignature,
+			msa_owner_key: T::AccountId,
+			msa_owner_proof: MultiSignature,
+			new_key: T::AccountId,
+			new_proof: MultiSignature,
 			add_key_payload: AddKeyData,
```

Comment:
I think I just noticed that we could remove the nonce field from AddKeyData. Maybe on a different task.

Reply:
Yep another task